JoyboyCommunity/src/shared/components/Post.tsx

```diff
@@ -42,6 +133,24 @@ interface PostProps {
 export const Post: React.FC<PostProps> = (props) => {
   const {post, event, repostedEvent} = props;
   const navigation = useNavigation<MainStackNavigationProps>();
+
+  const [isLiked, setIsLiked] = useState(false);
+  const [likes, setLikes] = useState(2);
+
+  const scale = useSharedValue(1);
+  const textOpacity = useSharedValue(likes > 0 ? 1 : 0);
+  const iconPosition = useSharedValue(likes > 0 ? 0 : 6); // Adjusted to account for the gap
+
+  const animatedIconStyle = useAnimatedStyle(() => ({
+    transform: [{scale: withTiming(scale.value, {duration: 25, easing: Easing.out(Easing.ease)})}],
+    marginLeft: withTiming(iconPosition.value, {duration: 25}), // Animation for moving the icon
```

Comment:
`useAnimatedStyle` should not include timing definitions, `withTiming` should be used when changing the shared value. This would definitely cause unexpected behaviour. For example it should be `scale.value = withTiming(1, {duration: 25))`.